packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -326,9 +326,32 @@ export class InterfaceVpcEndpointAwsService implements IInterfaceVpcEndpointServ
     const region = Lazy.uncachedString({
       produce: (context) => Stack.of(context.scope).region,
     });
-    this.name = `${prefix || 'com.amazonaws'}.${region}.${name}`;
+    const regionPrefix =Lazy.uncachedString({
+      produce: (context) => {
+        const regionName = Stack.of(context.scope).region;
+        return this.getDefaultEndpointPrefix(name, regionName);
+      },
+    });
+
+    this.name = `${prefix || regionPrefix}.${region}.${name}`;
     this.port = port || 443;
   }
+
+  /**
+   * Get the endpoint prefix for the service in the specified region
+   * because the prefix for some of the services in cn-north-1 and cn-northwest-1 are different
+   */
```

Comment:
Maybe we can add an AWS CLI sample in the doc comments showing how to get all endpoints from a region. That would be helpful.

Reply:
ok, I'll add the AWS CLI sample to the comments